File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    List<Integer> path = new LinkedList<>();
    List<List<Integer>> ans = new ArrayList<>();
    
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        backtracking(nums, 0);
        return ans;
    }
    
    private void backtracking(int[] nums, int startIdx) {
        ans.add(new ArrayList<>(path));
        
        for (int i = startIdx; i < nums.length; i++) {
            if (i > startIdx && nums[i] == nums[i - 1]) {
                continue;
            }
            
            path.add(nums[i]);
            backtracking(nums, i + 1);
            path.removeLast();
        }
    }
}

File: 0090-subsets-ii/0090-subsets-ii.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        ans = []
        nums.sort()

        def backtracking(path: List[int], startIdx: int) -> None:
            ans.append(path[:])

            for i in range(startIdx, len(nums)):
                if i > startIdx and nums[i] == nums[i - 1]:
                    continue

                path.append(nums[i])
                backtracking(path, i + 1)
                path.pop()
        
        backtracking([], 0)
        return ans